widget/view.controller.js

```diff
@@ -0,0 +1,200 @@
+                            }
+                        }
+                        else {
+                            if (result[config.slaMappedOn].itemValue == config.metItemvalue) {
+                                $interval.cancel($scope.timeinterval);
+                                $scope.sladate = new Date($filter('unixToDate')(result[config.sladate]));
+                                $scope.duedate = new Date($filter('unixToDate')(result[config.duedate]));
+                                $scope.createdate = new Date($filter('unixToDate')(result['createDate']));
+                                
+                                $scope.title = config.stopStateTitle;
+                                $scope.slastate = "Met";
+                                if (config.timeToDisplay === "consumedTime") {
+                                    stopCountDown('clockdiv', $scope.duedate, $scope.sladate, $scope.createdate);
+                                }
+                                else {
+                                    stopCountDown('clockdiv', $scope.duedate, $scope.sladate);
+                                }
+                            }
+                            else if (result[config.slaMappedOn].itemValue === config.pausedItemvalue) {
+                                $interval.cancel($scope.timeinterval);
+                                $scope.slastate = "Paused";
+                                $scope.title = "SLA Paused"
+                                $scope.pausedate = new Date($filter('unixToDate')(result[config.slapaused]));
+                                stopCountDown('clockdiv', $scope.duedate, $scope.pausedate);
+                            }
+                        }
+                    });
+                }
+            }
+            ).then(function (subscription) {
+                widgetsubscription = subscription;
+            });
+        }
+        widgetWSSubscribe();
+        function getTimeRemaining(duedate, slatime, consumedtime) {
+            var total = 0;
+            if (slatime && !consumedtime) {
+                total = Date.parse(duedate) - Date.parse(slatime);
+            }
+            else if (slatime && consumedtime) {
+                total = Date.parse(slatime) - Date.parse(consumedtime);
+            }
+            else {
+                total = Date.parse(duedate) - Date.parse(new Date());
+            }
+            if (total > 0) {
+                var seconds = Math.floor((total / 1000) % 60);
+                var minutes = Math.floor((total / 1000 / 60) % 60);
+                var hours = Math.floor((total / (1000 * 60 * 60)) % 24);
+                var days = Math.floor(total / (1000 * 60 * 60 * 24));
+                return {
+                    total,
+                    days,
+                    hours,
+                    minutes,
+                    seconds
+                };
+            }
+            else {
```

Comment:
Initialize these parameters once and add values in if block and return them once.